boot/zephyr/CMakeLists.txt

```diff
@@ -109,10 +109,12 @@ zephyr_library_sources(
   ${BOOT_DIR}/bootutil/src/image_rsa.c
   ${BOOT_DIR}/bootutil/src/image_ec256.c
   ${BOOT_DIR}/bootutil/src/image_ed25519.c
-  ${BOOT_DIR}/bootutil/src/bootutil_misc.c
```

Comment:
I don't think it makes sense to move all of this functionality into a library, especially one shared in other places.  Do we know what functions actually need to be used by the targeted image? I' thinking of something like `bootutil_buffer_is_erased` where the flash operations on the guest os might not even be the same.

Reply:
Ok, I can extract then these a few common functions to a common file and share only this file.

Reply:
`bootutil_buffer_is_erased()` uses OS primitives, so see no problem with it.
looks like only a few function are mcuboot's only:
boot_find_status()
boot_read_swap_size()
boot_write_swap_size(
boot_enc_key_off()
boot_write_enc_key()
boot_read_enc_key()
boot_write_copy_done()

I would move them to separate file.